spring-integration-core/src/test/java/org/springframework/integration/handler/MethodInvokingMessageProcessorTests.java

```diff
@@ -449,18 +448,6 @@ public void testProcessMessageCheckedException() throws Exception {
 				.withRootCauseInstanceOf(CheckedException.class);
 	}
 
-	@Test
-	public void testProcessMessageMethodNotFound() throws Exception {
-		TestDifferentErrorService service = new TestDifferentErrorService();
-		Method method = TestErrorService.class.getMethod("checked", String.class);
-		MethodInvokingMessageProcessor processor = new MethodInvokingMessageProcessor(service, method);
-		processor.setUseSpelInvoker(true);
-		processor.setBeanFactory(mock(BeanFactory.class));
-		assertThatExceptionOfType(MessageHandlingException.class)
-				.isThrownBy(() -> processor.processMessage(new GenericMessage<>("foo")))
-				.withCauseInstanceOf(SpelEvaluationException.class);
-	}
-
 	@Test
```

Comment:
Hmmm - I don't think this is right - there is no relationship between the methods on these two classes - it looks like `getMostSpecificMethod()` doesn't check that the method is in the class hierarchy - seems like a bug to me. I think this test should still pass.

Consider ignoring instead and perhaps open an issue in SF to see whether they consider the behavior to be correct?

Reply:
OK. Makes sense.
It bumped me as well: why `getMostSpecificMethod()` doesn't care about the class of source.

I'll revert the test change with an appropriate `@Ignore`.

We still don't pass for Kotlin variant, but that's also out of SI scope...

Reply:
Looks like it's intentional...

```java
return targetClass.getMethod(method.getName(), method.getParameterTypes());
```

But it seems odd to me.

Reply:
Here it is: https://github.com/spring-projects/spring-framework/issues/23824